core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -115,9 +115,13 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
     set("spark.master", master)
   }
 
-  /** Set a name for your application. Shown in the Spark web UI. */
+  /** Set a name for your application. Shown in the Spark web UI.
+    * For spark on kubernetes,Unify the three variables' name :
```

Comment:
It's not clear what this comment means, and the format needs to be fixed significantly.
You're also introducing a significant behavior change here, and seems like it's k8s specific? This part I don't think we can change.

Reply:
/** Set a name for your application. Shown in the Spark web UI.
    * For spark on kubernetes,Unify the three variables' name :
    *     pod name prefix in kubernetes / spark.app.name in spark ui  /  spark-app-name in pod's annotations
    * */

It uses the value of "-- name"  to name the pod name for spark on k8s when submitting tasks by spark-submit . Meanwhile KubernetesClientApplication calls start function ：
`start(args: Array[String], conf: SparkConf)`
in conf parameter, the value of  spark.app.name equals the value of "--name".
But the value of spark.app.name is not coincident.
`val conf = new SparkConf().**setAppName("JavaNetworkWordCount")`
`    _ui =
      if (conf.getBoolean("spark.ui.enabled", true)) {
        Some(SparkUI.create(Some(this), _statusStore, _conf, _env.securityManager, appName, "",
          startTime))
      } else {
        // For tests, do not enable the UI
        None
      }`

Create an UI by calling SparkUI's create function which contains _conf  parameter.
The value of spark.app.name is setting by calling setAppName method.
  
KubernetesClientApplication.start  and  SparkUI.create  ,both functions have SparkConf parameter. They have the possibility of inconsistency.

Refer to the pictures and content in the previous comments

Reply:
No, this isn't best solved by a new env variable. I'm not sure why you can't set the app name normally? I don't understand this.

This is also opened vs a branch, not master. it would have to be closed anyway. There's no JIRA either. But, I don't think this kind of change is correct.

Reply:
Thank you for your reply.